### ISSUE TITLE
[Org] Idris Developer Meeting Dec 2022

### ISSUE THREAD

**Issue by andrevidela** [idris-lang/Idris2#2773]
## Meta

This is an issue to keep track of the management of the Idris developer meeting, this is _not_ the announcement. We thought it might be nice to give more visibility to the community as to how the event is organised so that we can be more flexible to suggestion, kept accountable and train other people to organise such events.

## To do Pre

- [ ] post to mailing list
  - [ ] Proofread by someone
- [ ] post on the wiki 
  - [ ] Proofread by someone
- [ ] post on discord
    - [ ] Proofread by someone
- [ ] announce on twitter (who got access except edwin?)
	- [ ] Mastodon?
- [ ] setup discord role
- [ ] find a speaker for the first day
- [ ] make sure the links are available on discord
- [ ] Check if we have a good list of "good first first issue"
- [ ] Do we have a good way to teach people how to schedule the events?

## Todo on the day

- [ ] Unarchive the developer meeting channel
- [ ] Make an extra announcement using the new role

## Todo Post

- [ ] Archive the developer meeting channel
- [ ] prepare for release

## Info

Date: 05.12-09.12
location: Discord
Programme:
- Monday - 10am UTC+1 : kick off
- Friday - 3pm UTC+1: Show & tell + conclusion

### Discord Announcement

Dear Idris community,

We're gearing up toward the next Idris Developer meeting and we have picked a date: it will start 5th of December and end the 9th of december.

As the pandemic restriction ease in multiple countries we encourage you to create groups and meet together to create a hybrid event.

If this is your first Idris developer meeting, here is what you need to know:
- We attempt to use this week to hack, learn and socialise around the Idris community
- You are encouraged to show your work, project ideas and experiments with the language For this we've setup a calendar bot on which you can schedule a presentation. Usually those range from 10m to 1h.
- At the end we have a Show & Tell session where we share the worked that's been done and make a short demo.

### Wiki message

---

## Date

Monday 5th of December - Friday 9th of December 2022

## Aim

Spend some time hacking together and have a good time!

## Location

The event will take place on the [idris2 discord][DISCORD], social events will happen on [gather.town][GT] - the exact link will be shared on discord. No registration is required, just join the discord server and all the information will be shared on the following channels:
- #:loudspeaker: | announcements
- #:spiral_calendar: | events

## Format

The format will be similar to the previous IDM. There will be talks, discussions, coding sessions and time for socialising in the Idris Castle. The programme is yet to be determined, but will evolve and be updated here. If you have a suggestion for a talk you want to give, or have a discussion topic to propose, please add those below. Last time, talks happened in the middle of the week on Wednesday but there is no strong rule.

### Programme

- Monday 16/05/2022 10:00 UTC+1: Introduction
- Friday 20/05/2022 15:00 UTC+1: [Show & Tell](#show-and-tell-friday-1500-utc-1) + Conclusion

### Project ideas

Some things you might choose to work on during the week (please add any you can think of):

- A runtime debugger
- Help on the new core?

## Show and Tell (Friday 15:00 UTC +1)

Tell us what you did during the week!  Add your name here if you have something you want to show or contact one of the other moderators on discord, any time before the start.

## Participants/Registration

If you plan to attend and have joined the discord, but still want some form of "registration" you can list your name and/or github username below.

Please also consider adding your timezone, so that we can avoid scheduling too many things when people are asleep. 
In discord you can add it to your description. Participants are reminded that we expect the [Idris Community Standards](https://www.idris-lang.org/pages/community-standards.html) to be upheld throughout.

[DISCORD]: https://discord.gg/UX68fDs2jc
[GT]: gather.town

---

### Mailing list announcement

Dear Idris community,

We're gearing up toward the next Idris Developer meeting and we have picked a date: it will start 5th of December and end the 9th of December.

Like previous years, the event will take place on the [idris2 discord][DISCORD], social events will happen on [gather.town][GT] - the exact link will be shared on discord. No registration is required, just join the discord server and all the information will be shared on the following channels:
- #:loudspeaker: | announcements
- #:spiral_calendar: | events

As the pandemic restriction ease in multiple countries we encourage you to create groups and meet together to create a hybrid event.

If this is your first Idris developer meeting, here is what you need to know:
- We attempt to use this week to hack, learn and socialise around the Idris community
- You are encouraged to show your work, project ideas and experiments with the language For this we've setup a calendar bot on which you can schedule a presentation. Usually those range from 10m to 1h.
- At the end we have a Show & Tell session where we share the worked that's been done and make a short demo.

**Comment by andrevidela** [comment]
@CodingCellist @gallais Can you proof-read/make sure I didn't forget anything? Iwas thining of making the announcements on monday since people might not (should not) be looking at their notifications on sunday

**Comment by CodingCellist** [comment]
@andrevidela, some comments:
- `who got access` -> `who's got access`  (side-note: I do! : ) )
- `good first first issue` -> `good first issue` (duplicate `first`)
- `make sure the links are available on discord` -- what links? elaborate
- Do we want to a release after each IDM?
- `Date` -> `Duration` / `Dates` ?
- > We're gearing up toward the next Idris Developer meeting and we have picked a date: it will start 5th of December and end the 9th of december.  
  
  I would reword this as  
  
  > We're gearing up for the next Idris Developer Meeting (IDM) and have picked a date: the IDM will start on Monday the 5th of December and last the week, ending on Friday the 9th of December.`  -- I think this sounds better, but feel free to adjust.
- `create a hybrid event.` -> `create a hybrid event where restrictions permit and if people feel comfortable.`  -- I prefer the extra caution in this wording.
- `experiments with the language For this` -> `experiments with the language. For this` (missing full stop)
- `Monday 5th` -> `Monday the 5th` (and similar for Friday on that line)
- `gather.town` needs to be a proper hyperlink, otherwise GitHub tries to redirect to an issue authored by that (non-existent) person ^^
- `socialising in the Idris Castle` -- add a note saying this is on [gather.town](https://gather.town) to avoid confusion for newcomers
- `Help on the new core?` -- we should probably double-check with Edwin if he's okay with this and what kind of contributions are welcome (if any)  : )

Most of these comments also apply to the mailing list section.

**Comment by andrevidela** [comment]
We have the two other pages for the previous developer meetings up on the wiki, should we remove them?

For reference, those are the pages https://github.com/idris-lang/Idris2/wiki/%5BArchive%5D-Idris-Developer-Meeting,-May-2022 and https://github.com/idris-lang/Idris2/wiki/%5BArchive%5D-Idris-Developers-Meeting,-December-2021

**Comment by CodingCellist** [comment]
I kept them as a sort of archive; I ~~have an irrational fear of deleting stuff~~ like having a record. But I don't know if it's relevant info?  : )

**Comment by andrevidela** [comment]
Given that the wiki is it's own git repository, deleting them is not going to be permanent, we can always get back to them. I think it would make things a bit more tidy

**Comment by andorp** [comment]
Project ideas:
+ Applicative Do notation as in: https://github.com/purescript/purescript/issues/2435#issuecomment-268023442
